test/components/Notification/BasicNotification.test.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { expect, it, beforeEach, afterEach, vi } from 'vitest';
+import { act } from 'react-dom/test-utils';
+import BasicNotification from '../../../src/components/Notification/BasicNotification';
+import { NotificationContext } from '../../../src/contexts/NotificationContext';
+
+it('renders correctly', () => {
+  render(
+    <BasicNotification
+      severity="success"
+      message="my test message"
+      id={123456}
+      dataTestId="test-id"
+    />
+  );
+
+  expect(screen.getByText('my test message')).not.toBeNull();
+});
+
+beforeEach(() => {
+  vi.useFakeTimers();
+});
+
+afterEach(() => {
+  vi.useRealTimers();
+});
+
+it('calls remove after 8 seconds', () => {
+  const remove = vi.fn();
+
+  render(
+    <NotificationContext.Provider value={{ remove }}>
+      <BasicNotification
+        severity="success"
+        message="my test message"
+        id={123456}
+        dataTestId="test-id"
+      />
+    </NotificationContext.Provider>
+  );
+
+  act(() => {
+    vi.advanceTimersByTime(8000);
+  });
+
+  expect(remove).toHaveBeenCalledOnce();
+});
+
+it('calls remove with the correct id', () => {
```

Comment:
This test looks really good as is. Nice work. 

The `act` call is an interesting one, but seems to be doing exactly what you intend, so I'm fine with it.

You may also consider looking over react testing library's `waitFor` feature:
https://testing-library.com/docs/dom-testing-library/api-async/

They may be easier to work with than `act`